### PR TITLE
Center single page PDF on load

### PR DESCRIPTION
**Pull request by josephineb**
## Pull Request Checklist

- [x] I have run `yarn build` to ensure that the project builds successfully.
- [x] I have updated `CHANGELOG.md` with the necessary changes made in this pull request.

## Description

When a single page PDF is uploaded, it is automatically centered in the PDF viewer.

## Related Issues

#278

### CODE REVIEW

**Comment by courtneypattison** [comment]
Looks great! Thank you @josephineb!

Since we only recenter when there's one page, the layoutWidth and layoutHeight are just the width and height of the page. Therefore, we can use those sizes on page load success. Something like this:
```
function onPageLoadSuccess(pdfProxy: PDFPageProxy) {
    const scale = (pdfProxy.userUnit || 1) * PDF_TO_CSS_UNITS;
    const width = pdfProxy.view[2] * scale;
    const height = pdfProxy.view[3] * scale;
    setPageSize({
      action: "setPage",
      pageNumber: pdfProxy.pageNumber,
      width,
      height,
    });
    if (pageCount === 1) {
      transformer.recenter(gridCenter, width, height);
    }
  }
```

That way we don't need any additional state! What do you think?

**Comment by courtneypattison** [comment]
It's a bit of a special case. We may need to add the state back in the future but for now I think it's a bit simpler. Thank you so much for the quick fix! Would you be able to remove `transformer` and `grid center` from the `useEffect` as well please since those aren't being used anymore. Then I can merge this PR in :)

**Comment by josephineb** [comment]
> It's a bit of a special case. We may need to add the state back in the future but for now I think it's a bit simpler. Thank you so much for the quick fix! Would you be able to remove `transformer` and `grid center` from the `useEffect` as well please since those aren't being used anymore. Then I can merge this PR in :)

Ah, I didn't clean that up. I think it's good to go now ;)

File: app/_components/pdf-viewer.tsx [hunk 1]
```diff
@@ -72,12 +75,17 @@ export default function PdfViewer({
 
   function onPageLoadSuccess(pdfProxy: PDFPageProxy) {
     const scale = (pdfProxy.userUnit || 1) * PDF_TO_CSS_UNITS;
+    const width = pdfProxy.view[2] * scale;
+    const height = pdfProxy.view[3] * scale;
     setPageSize({
       action: "setPage",
       pageNumber: pdfProxy.pageNumber,
       width: pdfProxy.view[2] * scale,
       height: pdfProxy.view[3] * scale,
```
Comment (Nifdee) [inline app/_components/pdf-viewer.tsx:84]: Super nit, but we can reuse the `width` and `height` values from above :)
```suggestion
      width,
      height,
```

Comment (josephineb) [reply app/_components/pdf-viewer.tsx]: > Super nit, but we can reuse the `width` and `height` values from above :)

You're absolutely right! I have made the change. I apologize for the late reply; I was on holiday and unable to address this sooner. I also rebased the branch, resolving some conflicts. Please review it to ensure my changes are still correct :smiley:

**Comment by courtneypattison** [comment]
Sorry for not merging this earlier! I read that nit and then left this PR and forgot. It must've been annoying fixing the merge conflicts. Thanks again! It's going to be really nice not having the recenter with large PDFs.

**Pull request merged**